qemu/tests/block_hotplug.py

```diff
@@ -121,42 +126,47 @@ def get_new_disk(disk1, disk2):
 
             device_list.append(device)
             if not new_disks:
-                raise exceptions.TestFail("Cannot find new disk after hotplug.")
+                test.fail("Cannot find new disk after hotplug.")
 
             if params.get("need_plug") == "yes":
                 disk = new_disks[0]
             else:
                 disk = new_disks[num]
 
+            session = vm.wait_for_login(timeout=timeout)
+            if params.get("format_disk", "no") == "yes":
+                if iteration == 0:
+                    error_context.context("Format disk", logging.info)
+                    utils_misc.format_windows_disk(session,
+                                                   disk_index[num],
+                                                   mountpoint=disk_letter[num])
             error_context.context("Check block device after hotplug.",
                                   logging.info)
-            if params.get("pci_test_cmd"):
+            if params.get("disk_op_cmd"):
                 if params.get("os_type") == "linux":
-                    test_cmd = params.get("pci_test_cmd") % (disk, disk)
+                    test_cmd = params.get("disk_op_cmd") % (disk, disk)
+                elif params.get("os_type") == "windows":
+                    test_cmd = params.get("iozone_cmd") % (disk_letter[num],
+                                                           disk_letter[num])
                 else:
-                    test_cmd = re.sub("PCI_NUM", "%s" % (num + 1),
-                                      params.get("pci_test_cmd"))
-                session = vm.wait_for_login(timeout=timeout)
-                s, o = session.cmd_status_output(test_cmd, timeout=disk_op_timeout)
-                session.close()
-                if s:
-                    raise exceptions.TestFail("Check for block device failed "
-                                              "after hotplug, Output: %r" % o)
+                    error_context.context("Has not implemented.", logging.info)
+                    raise NotImplementedError
+                status, output = session.cmd_status_output(test_cmd, timeout=disk_op_timeout)
+                if status:
+                    test.fail("Check for block device failed "
+                              "after hotplug, Output: %r" % output)
+            session.close()
 
         sub_type = params.get("sub_type_after_plug")
         if sub_type:
             error_context.context(context_msg % (sub_type, "after hotplug"),
                                   logging.info)
             utils_test.run_virt_sub_test(test, params, env, sub_type)
-
-        sub_type = params.get("sub_type_before_unplug")
-        if sub_type:
-            error_context.context(context_msg % (sub_type, "before unplug"),
-                                  logging.info)
-            utils_test.run_virt_sub_test(test, params, env, sub_type)
+            if sub_type == "shutdown":
+                return
```

Comment:
The framework does not always shutdown the guest as your want, so please perform the shutdown action explicitly.

Reply:
thanks, change to judge vm.is_dead()